### PR TITLE
[move-prover] Update touched addresses to include function arguments

### PR DESCRIPTION
**Pull request by ckaiwxm**
## Motivation

- This is a follow-up PR for #219 to add function arguments that has an address or a signer field to the touched addresses

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?

Yes.

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

### CODE REVIEW

File: language/move-prover/interpreter/src/concrete/player.rs [hunk 1]
```diff
@@ -109,6 +109,21 @@ impl<'env> FunctionContext<'env> {
...
+        match val {
+            BaseValue::Address(v) | BaseValue::Signer(v) => {
+                addresses.push(*v);
+            }
+            BaseValue::Vector(vec) | BaseValue::Struct(vec) => {
+                for (_, val) in vec.iter().enumerate() {
+                    Self::collect_addresses(val, addresses);
+                }
+            }
+            _ => (),
```
Comment (meng-xu-cs) [inline language/move-prover/interpreter/src/concrete/player.rs:123]: Probably a good idea to fully expand other variants of the `BaseValue` enum.

File: language/move-prover/interpreter/src/concrete/player.rs [hunk 2]
```diff
@@ -117,6 +132,14 @@ impl<'env> FunctionContext<'env> {
         global_state: &mut GlobalState,
         eval_state: &mut EvalState,
     ) -> ExecResult<LocalState> {
+        // collect addresses
+        for (_, typed_arg) in typed_args.iter().enumerate() {
+            let mut addresses: Vec<AccountAddress> = vec![];
```
Comment (meng-xu-cs) [inline language/move-prover/interpreter/src/concrete/player.rs:137]:
```suggestion
            let mut addresses = vec![];
```

File: language/move-prover/interpreter/src/concrete/value.rs [hunk 3]
```diff
@@ -1139,6 +1139,11 @@ impl GlobalState {
         &self.touched_addresses
     }
 
+    /// Put new addresses to all the addresses that are touched by the bytecode
+    pub fn put_touched_addresses(&mut self, addresses: &Vec<AccountAddress>) {
```
Comment (meng-xu-cs) [inline language/move-prover/interpreter/src/concrete/value.rs:1143]:
```suggestion
    pub fn put_touched_addresses(&mut self, addresses: &[AccountAddress]) {
```

**Review by meng-xu-cs — COMMENTED** [review]
Thanks for the changes! Just some minor fix ups and the PR is good to go

File: language/move-prover/interpreter/src/concrete/player.rs (hunk 1 above)
Comment (ckaiwxm) [reply language/move-prover/interpreter/src/concrete/player.rs]: Make sense! Refactored

**Review by meng-xu-cs — APPROVED** [review]
LGTM

**Pull request merged**